libtransmission/tr-lpd.cc

```diff
@@ -244,6 +248,71 @@ class tr_lpd_impl final : public tr_lpd
     }
 
 private:
+#ifdef _WIN32
+    void GetIPv4SourceAddress(sockaddr_in *source_addr)
+    {
+        int i;
+        PMIB_IPADDRTABLE pIPAddrTable;
+        DWORD dwSize = 0;
+        DWORD dwRetVal = 0;
+
+        memset(source_addr, 0, sizeof(sockaddr_in));
+        source_addr->sin_addr.s_addr = INADDR_ANY;
+
+        pIPAddrTable = (MIB_IPADDRTABLE *) HeapAlloc(GetProcessHeap(), 0, sizeof (MIB_IPADDRTABLE));
```

Comment:
Does this need to be a raw unmanaged pointer?

There are still other raw unmanaged pointers in libtransmission but we're generally trying to remove their use rather than add them.

What if this was a `std::vector<std::byte>(MIB_IPADDRTABLE)` instead, and then the HeapFree + HeapAlloc pair on lines 267-268 could be replaced with a `.resize()`

Reply:
Cool ... lemmie check into that.